File: freecodecamp/daily-code-challenge/2026-03-29.py
def is_valid_isbn10(s):

    s = s.replace("-", "")

    if len(s) != 10:
        return False

    total = 0

    for i in range(10):
        ch = s[i]

        if i == 9 and ch == 'X':
            value = 10
        elif ch.isdigit():
            value = int(ch)
        else:
            return False

        total += value * (i + 1)

    return total % 11 == 0

File: freecodecamp/daily-code-challenge/2026-03-29.js
function isValidIsbn10(str) {

  let s = str.replace(/-/g, "");

  // Length check
  if (s.length !== 10) return false;

  let sum = 0;

  for (let i = 0; i < 10; i++) {
    let ch = s[i];
    let value;

    if (i === 9 && ch === 'X') {
      value = 10;
    } else if (ch >= '0' && ch <= '9') {
      value = ch.charCodeAt(0) - '0'.charCodeAt(0);
    } else {
      return false;
    }

    sum += value * (i + 1);
  }

  return sum % 11 === 0;
}